src/app/donation-start/donation-start-form/donation-start-form.component.html

```diff
@@ -444,7 +444,7 @@ <h3 class="b-rt-0 b-m-0 b-bold"><span class="span-hr"></span>How does Big Give u
             class="continue b-w-100 b-rt-0"
             mat-raised-button
             color="primary"
-            [disabled]="(!stripePaymentMethodReady && !selectedSavedMethod) || !paymentGroup.valid"
+            [disabled]="! readyToProgressFromPaymentStep"
```

Comment:
in the old version this line was confusing because `stripePaymentMethodReady` was set for several unrelated reasons in different scenarios. The logic is now I think much clearer in the implementation of `PaymentReadinessTracker.readyToProgressFromPaymentStep`.

Whether it was necessary to refactor step by step to get there is debatable - maybe we should have just written that from scratch. But this way we can be confident that we're preserving the behaviour.

Reply:
Very nice! 🙌